include/dynd/types/tuple_type.hpp

```diff
@@ -145,13 +147,13 @@ namespace ndt {
     }
 
     /** Makes a tuple type with the specified types */
-    static type make(const nd::array &field_types, bool variadic = false)
+    static type make(std::vector<type> field_types, bool variadic = false)
     {
       return type(new tuple_type(field_types, variadic), false);
```

Comment:
If I understand you correctly, constructing tuples can also be `const std::vector &`, rather then `const std::vector`. It just can't be stored as a reference `&. So the constructor can take a ` const std::vector &`, but store it as a `std::vector` or `const std::vector`.